internal/regexp/regexp.go

```diff
@@ -0,0 +1,76 @@
+package regexp
+
+import (
+	"fmt"
+	"reflect"
+	"regexp"
+	"strings"
+)
+
+// Compile takes a regular expression and a custom lookup struct that should match the named groups
+// in the regexp.
+//
+// `lookup` should always be a pointer to a struct with only public `int` properties.
+//
+// Example:
+// var lookup = struct {
+//   FirstPart int
+//   SecondPart int
+// }{}
+// var re, err = regexp.Compile(``, &lookup) // Handle err or use MustCompile
```

Comment:
Unusual variable decleration. Couldn't this be the following instead?

```suggestion
// re, err := regexp.Compile(``, &lookup) // Handle err or use MustCompile
```